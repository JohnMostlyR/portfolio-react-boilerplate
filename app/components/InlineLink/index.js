import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { theme } from '../../styles/theme';

const { color, textShadowColor } = theme.inlineLink;

const InlineLink = styled(Link)`
  background-color: transparent;
  background-image: linear-gradient(currentColor, currentColor);
  background-origin: padding-box;
  background-position: center bottom 10%;
  background-repeat: no-repeat;
  background-size: 100% 1.5px;
  color: ${color};
  cursor: pointer;
  text-decoration: none;
  text-shadow: 3px 0 ${textShadowColor}, 2px 0 ${textShadowColor},
    1px 0 ${textShadowColor}, -1px 0 ${textShadowColor},
    -2px 0 ${textShadowColor}, -3px 0 ${textShadowColor};
  touch-action: manipulation;
  transition: color 0.1s ease-out;

  @supports (text-decoration-skip-ink: auto) {
    background: none;
    text-decoration: underline;
    text-decoration-skip-ink: auto; /* stylelint-disable-line property-no-unknown */
    text-shadow: none;
  }
`;

InlineLink.propTypes = {
  to: PropTypes.string.isRequired,
};

export default InlineLink;
