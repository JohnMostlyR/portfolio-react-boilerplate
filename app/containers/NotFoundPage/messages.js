/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'portfolio.page.NotFoundPage.pageTitle',
    defaultMessage: 'NOT FOUND',
  },
  pageDescription: {
    id: 'portfolio.page.NotFoundPage.pageDescription',
    defaultMessage:
      'The page you requested does not exist or has been removed. Sorry about that',
  },
  title: {
    id: 'portfolio.page.NotFoundPage.title',
    defaultMessage:
      'Johan Meester : Portfolio : Front-End Developer : {pageTitle}',
  },
  content: {
    id: 'portfolio.page.NotFoundPage.content',
    defaultMessage: `I am sorry, but the page you requested does not exist or has been removed.
      Please go to the {home} page or, if you think this is an error, to the {contact} page and contact me.`,
  },
  home: {
    id: 'portfolio.page.NotFoundPage.home',
    defaultMessage: 'start',
  },
  contact: {
    id: 'portfolio.page.NotFoundPage.contact',
    defaultMessage: 'contact',
  },
});
