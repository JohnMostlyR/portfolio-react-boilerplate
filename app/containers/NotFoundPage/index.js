/**
 * NotFoundPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Helmet } from 'react-helmet';

import messages from './messages';
import PageContent from '../../components/PageContent';
import InlineLink from '../../components/InlineLink';

export default class NotFound extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <React.Fragment>
        <Helmet>
          <title>Niet gevonden</title>
          <meta name="description" content="Niet gevonden pagina van Johan Meester zijn portfolio" />
        </Helmet>
        <PageContent
          title={<FormattedMessage {...messages.title} />}
          content={
            <FormattedMessage
              {...messages.content}
              values={{
                home: (
                  <FormattedMessage {...messages.home}>
                    {
                      (message) => <InlineLink to={'/'}>{message}</InlineLink>
                    }
                  </FormattedMessage>
                ),
                contact: (
                  <FormattedMessage {...messages.contact}>
                    {
                      (message) => <InlineLink to={'/contact'}>{message}</InlineLink>
                    }
                  </FormattedMessage>
                ),
              }}
            />
          }
        />
      </React.Fragment>
    );
  }
}
