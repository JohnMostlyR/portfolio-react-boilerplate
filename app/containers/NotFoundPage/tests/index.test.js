import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'react-router-redux';
import createHistory from 'history/createMemoryHistory';

import NotFoundPage from '../index';

import InlineLink from '../../../components/InlineLink';
import configureStore from '../../../configureStore';

describe('<NotFoundPage />', () => {
  const initialState = {};
  const history = createHistory();
  const store = configureStore(initialState, history);

  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(
      <NotFoundPage />
    );

    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

  it('should have a link to the home page', () => {
    const renderedComponent = mount(
      <Provider store={store}>
        <IntlProvider locale={'en'}>
          <ConnectedRouter history={history}>
            <NotFoundPage />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>
    );

    const linkToHomePage = renderedComponent.find(<InlineLink to={'/'} />);
    expect(renderedComponent.contains(InlineLink)).toEqual(true);
    expect(linkToHomePage).toBeDefined();
  });

  it('should have a link to the contact page', () => {
    const renderedComponent = mount(
      <Provider store={store}>
        <IntlProvider locale={'en'}>
          <ConnectedRouter history={history}>
            <NotFoundPage />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>
    );

    const linkToHomePage = renderedComponent.find(<InlineLink to={'/contact'} />);
    expect(renderedComponent.contains(InlineLink)).toEqual(true);
    expect(linkToHomePage).toBeDefined();
  });
});
