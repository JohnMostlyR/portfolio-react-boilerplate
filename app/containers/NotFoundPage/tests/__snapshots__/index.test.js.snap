// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render and match the snapshot 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Niet gevonden
    </title>
    <meta
      content="Niet gevonden pagina van Johan Meester zijn portfolio"
      name="description"
    />
  </HelmetWrapper>
  <PageContent
    content={
      <FormattedMessage
        defaultMessage="I am sorry, but the page you requested does not exist or has been removed.
            Please go to the {home} page or, if you think this is an error, to the {contact} page and contact me."
        id="portfolio.page.NotFoundPage.content"
        values={
          Object {
            "contact": <FormattedMessage
              defaultMessage="contact"
              id="portfolio.page.NotFoundPage.contact"
              values={Object {}}
          >
              [Function]
          </FormattedMessage>,
            "home": <FormattedMessage
              defaultMessage="start"
              id="portfolio.page.NotFoundPage.home"
              values={Object {}}
          >
              [Function]
          </FormattedMessage>,
          }
        }
      />
    }
    title={
      <FormattedMessage
        defaultMessage="NOT FOUND"
        id="portfolio.page.NotFoundPage.title"
        values={Object {}}
      />
    }
  />
</React.Fragment>
`;
